### DISCUSSION TITLE
Possible to Install With a Different Python Version using Pyenv?

### DISCUSSION THREAD

**Discussion by PresTrembleyIIIEsq** [bmaltais/kohya_ss#1643] [General]
Hello,

I'm on Manjaro, and I have Python 3.11.x installed system-wide, and I'd like to not mess that up at all.

I'm hoping to use `pyenv` to install Kohya using Python 3.10.x in the venv, but everything I've tried still just leads to the following message when running setup.sh:

`Incompatible Python version: 3.11.x required 3.[9, 10]`

Should this be possible to do, and if so, does anyone have any kind of beginner-friendly (or at least intermediate) guide on installing Kohya this way?

**Comment by itsmeherefolks** [comment]
Hi @PresTrembleyIIIEsq 

for me pyenv works on Manjaro with the following steps:

- Install pyenv from the official repository
- Installation of the desired python version: `pyenv install 3.10.13`
- Set python version for the kohya directory `cd .../kohya_ss` and `pyenv local 3.10.13`
- Initialization of pyenv in the shell. Add the following lines (e.g. with `nano ~/.zshrc`)

```
if command -v pyenv 1>/dev/null 2>&1; then
  eval "$(pyenv init -)"
fi
```
